### PR TITLE
[CH 10275] Add Support for Variations and Sorting in Search/Browse Responses

### PR DESCRIPTION
**Pull request by jjl014**
All tests pass 👍 

#### Updates:
* Add Search Response test resource for variations
* Update Browse Response test resource with additional metadata/sort options
* Add support for sort options to Search and Browse response models
* Add support for variations to Result models
* Update `moveMetadataOutOfResultData` to handle variations as well

### CODE REVIEW

File: constructorio-client/src/test/java/io/constructor/client/ConstructorIOBrowseTest.java [hunk 1]
```diff
@@ -92,6 +127,37 @@ public void BrowseShouldReturnAResultWithBrandFacets() throws Exception {
         assertTrue("browse result id exists", response.getResultId() != null);
     }
 
+    @Test
+    public void SearchShouldReturnAResultWithVariations() throws Exception {
```
Comment (rubikzube) [inline constructorio-client/src/test/java/io/constructor/client/ConstructorIOBrowseTest.java:131]: Should this be `BrowseShouldReturnAResultWithVariations`?

File: constructorio-client/src/test/java/io/constructor/client/ConstructorIOBrowseTest.java [hunk 2]
```diff
@@ -92,6 +127,37 @@ public void BrowseShouldReturnAResultWithBrandFacets() throws Exception {
...
+        assertEquals("browse result variation [facets] exists", response.getResponse().getResults().get(0).getVariations().get(0).getData().getFacets().size(), 8);
+        assertEquals("browse result variation [value] exists", response.getResponse().getResults().get(0).getVariations().get(0).getValue(), "Coat Aspesi blue");
+        assertEquals("browse result variation [variation id] exists", response.getResponse().getResults().get(0).getVariations().get(0).getData().getVariationId(), "M0E20000000ECTT");
+        assertEquals("browse result variation [url] exists", response.getResponse().getResults().get(0).getVariations().get(0).getData().getUrl(), "https://demo.commercetools.com/en/aspesi-coat-I502997385098-blue.html");
+        assertEquals("browse results count as expected", (int) response.getResponse().getTotalNumberOfResults(), 562);
+        assertTrue("browse result id exists", response.getResultId() != null);
+    }
+
+    @Test
+    public void SearchShouldReturnAResultWithSortOptions() throws Exception {
```
Comment (rubikzube) [inline constructorio-client/src/test/java/io/constructor/client/ConstructorIOBrowseTest.java:147]: Should this be `SearchShouldReturnAResultWithSortOptions `?

File: constructorio-client/src/test/java/io/constructor/client/ConstructorIOBrowseTest.java (hunk 1 above)
Comment (jjl014) [reply constructorio-client/src/test/java/io/constructor/client/ConstructorIOBrowseTest.java]: yup, thanks for catching that. updated

File: constructorio-client/src/test/java/io/constructor/client/ConstructorIOBrowseTest.java (hunk 2 above)
Comment (jjl014) [reply constructorio-client/src/test/java/io/constructor/client/ConstructorIOBrowseTest.java]: yup, thanks for catching that. updated

**Pull request merged**